### PR TITLE
Update examples to work with optimistic DA

### PR DESCRIPTION
**Pull request by kmchicoine**

### CODE REVIEW

**Review by elliedavidson — APPROVED** [review]
Lgtm!

**Comment by DieracDelta** [comment]
Nice! One small nit: could we rename `modDA.rs` to `mod_da.rs` to keep it consistent with the other file namings?

File: examples/infra/modDA.rs [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+        );
+        let hotshot = HotShot::init(
+            pk,
+            sk,
+            config.node_index,
+            config.config,
+            MemoryStorage::empty(),
+            exchanges,
+            initializer,
+            NoMetrics::boxed(),
```
Comment (DieracDelta) [inline examples/infra/modDA.rs:232]: Do we have any plans when we benchmark to include metrics?

File: examples/infra/modDA.rs [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+    /// Initializes networking, returns self
+    async fn initialize_networking(
+        config: NetworkConfig<TYPES::SignatureKey, TYPES::ElectionConfigType>,
+    ) -> Self;
+
+    /// Initializes the genesis state and HotShot instance; does not start HotShot consensus
+    /// # Panics if it cannot generate a genesis block, fails to initialize HotShot, or cannot
+    /// get the anchored view
+    /// Note: sequencing leaf does not have state, so does not return state
+    async fn initialize_state_and_hotshot(&self) -> HotShotHandle<TYPES, NODE> {
```
Comment (DieracDelta) [inline examples/infra/modDA.rs:181]: It's awesome how this is a default implementation.

File: examples/web-server-da/orchestrator.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#[cfg_attr(
+    feature = "tokio-executor",
+    tokio::main(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(feature = "async-std-executor", async_std::main)]
+#[instrument]
+async fn main() {
+    let args = OrchestratorArgs::parse();
+
+    run_orchestrator_da::<SDemoTypes, ThisMembership, DANetwork, QuorumNetwork, NodeImpl>(args)
```
Comment (DieracDelta) [inline examples/web-server-da/orchestrator.rs:26]: Slick!!

File: examples/web-server-da/orchestrator.rs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#[path = "../infra/modDA.rs"]
+pub mod infra_da;
+
+#[cfg_attr(
+    feature = "tokio-executor",
+    tokio::main(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(feature = "async-std-executor", async_std::main)]
+#[instrument]
+async fn main() {
```
Comment (DieracDelta) [inline examples/web-server-da/orchestrator.rs:23]: nit (maybe this is done elsewhere?): might be worth running `setup_logging` and `seup_backtrace`.

File: examples/infra/modDA.rs (hunk 1 above)
Comment (elliedavidson) [reply examples/infra/modDA.rs:232]: Ideally we would, but realistically I don't think we'll have the time to do it.

**Pull request merged**